lib/connectors_async/job.rb

```diff
@@ -86,5 +86,9 @@ def is_finished?
 
       [ConnectorsShared::JobStatus::FINISHED, ConnectorsShared::JobStatus::FAILED].include?(status)
     end
+
+    def should_wait?
+      @data[:documents].length > 500
```

Comment:
Let's pull this `500` out into `ConnectorsShared::Constants` or configuration or something, just to make it easier to find and update in the future.